Scripts/Runtime/Entities/TaskSystem/TaskFlow/Node/DataStreamNode.cs

```diff
@@ -18,27 +18,17 @@ internal class DataStreamNode : AbstractNode
                               TaskFlowGraph taskFlowGraph,
                               AbstractTaskSystem taskSystem,
                               AbstractTaskDriver taskDriver,
-                              AbstractTaskStream taskStream,
-                              bool isResolveTarget) : base(taskFlowGraph, taskSystem, taskDriver)
+                              AbstractTaskStream taskStream) : base(taskFlowGraph, taskSystem, taskDriver)
         {
             m_Lookup = lookup;
             DataStream = dataStream;
             TaskStream = taskStream;
-            IsResolveTarget = isResolveTarget;
+            IsResolveTarget = taskStream.IsDataStreamAResolveTarget && taskStream.GetDataStream() == dataStream;
```

Comment:
Just so I understand. It's possible to pass a task stream that's unrelated to the data stream? It just won't be a resolve target?

Reply:
A TaskStream can contain one or more DataStreams. 

You as the consuming developer can say that a TaskStream is a ResolveTarget and that will ensure that the primary DataStream is used as the ResolveTarget under the hood. 

But you could also say that the TaskStream is Cancellable so it will have a secondary DataStream in it. That DataStream is not a ResolveTarget. 

This check and organization is bit gross but it does the job today. I've added the TODO to ensure this is looked at for #66, #67 or #68